packages/realm/src/Object.ts

```diff
@@ -58,14 +61,22 @@ const DEFAULT_PROPERTY_DESCRIPTOR: PropertyDescriptor = { configurable: true, en
 /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
 const PROXY_HANDLER: ProxyHandler<RealmObject<any>> = {
   ownKeys(target) {
-    return Reflect.ownKeys(target).concat([...target.keys()].map(String));
+    return Reflect.ownKeys(target).concat(target[KEY_ARRAY]);
   },
   getOwnPropertyDescriptor(target, prop) {
-    if (typeof prop === "string" && target.keys().includes(prop)) {
+    if (typeof prop === "string" && target[KEY_SET].has(prop)) {
       return DEFAULT_PROPERTY_DESCRIPTOR;
-    } else {
-      return Reflect.getOwnPropertyDescriptor(target, prop);
     }
+    const res = Reflect.getOwnPropertyDescriptor(target, prop);
```

Comment:
I like the less abbreviated variant.

```suggestion
    const result = Reflect.getOwnPropertyDescriptor(target, prop);
```